python/pyarrow/parquet.py

```diff
@@ -1491,6 +1503,16 @@ def __init__(self, path_or_paths, filesystem=None, filters=None,
                     "Keyword '{0}' is not yet supported with the new "
                     "Dataset API".format(keyword))
 
+        if (
+            hasattr(path_or_paths, "__fspath__") and
+            filesystem is not None and
+            not _is_local_file_system(filesystem)
+        ):
+            raise TypeError(
+                "Path-like objects with __fspath__ must only be used with "
+                f"local file systems, not {type(filesystem)}"
```

Comment:
If you move this check a bit below, after the validation of filesystem keyword (which calls `_ensure_filesystem`, and this will map an fsspec LocalFileSystem to our own LocalFileSystem), then you don't need to check for fsspec, but only for our own LocalFileSystems.